File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/MoveTool.java
package com.savvasdalkitsis.gameframe.draw.model;

public class MoveTool extends AbstractDrawingTool {

    @Override
    public void drawOn(Layer layer, int startColumn, int startRow, int column, int row, int color) {
        if (layer.isBackground()) {
            return;
        }
        layer.getColorGrid().translate(column - startColumn, row - startRow);
    }

    @Override
    public void finishStroke(Layer layer) {
        layer.getColorGrid().freezeTranslation();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/draw/model/MoveTool.kt
package com.savvasdalkitsis.gameframe.draw.model

class MoveTool : AbstractDrawingTool() {

    override fun drawOn(layer: Layer, startColumn: Int, startRow: Int, column: Int, row: Int, color: Int) {
        if (layer.isBackground) {
            return
        }
        layer.colorGrid.translate(column - startColumn, row - startRow)
    }

    override fun finishStroke(selectedLayer: Layer) {
        selectedLayer.colorGrid.freezeTranslation()
    }
}
